woocommerce/woocommerce-ios
# Shipping Labels: Validate that address name field is not empty

Fixes: #4480

## Description

When you start to create a shipping label, we auto-fill the origin address name field with the first and last name from the user's account settings. If the account settings are not in storage, or don't include a first/last name, the name is left blank. This causes problems later on, because the name is required (e.g. to load carriers and rates for the package).

This PR adds local validation to check that the name field is not empty. If it is empty, it opens the Edit Address screen to prompt the user to enter a name. (We already have validation on the Edit Address screen to ensure the name field is filled out there.)

Note that we have related issues for remote name validation (https://github.com/Automattic/woocommerce-services/issues/2457) and making sure account settings are synced in storage after login (https://github.com/woocommerce/woocommerce-ios/issues/4600), which aren't part of this PR.

## Changes

* Adds a check for `address.name.isEmpty` before doing the remote address validation in `ShippingLabelFormViewModel`.
* Adds a unit test.

Before|After
-|-
![validate-name-before](https://user-images.githubusercontent.com/8658164/125477810-3d78cc42-614e-4abc-88d4-b0eb6c773dbf.gif)|![validate-name-after](https://user-images.githubusercontent.com/8658164/125477822-de7332ac-6ec0-49cd-8c0c-698b0ce40c66.gif)




## Testing

1. Make sure your `First name` and `Last name` are empty in your [WordPress.com account settings](https://wordpress.com/me).
2. Make sure you have configured the WooCommerce WCShip plugin, and you have one or more payment methods set up in WP Admin under WooCommerce > Settings > Shipping > WooCommerce Shipping.
3. Launch the app.
4. If you are already logged in, log out and log in again (to make sure you don't have old account settings stored in the app).
5. Open an order detail that is eligible for the shipping label.
6. Tap the button "Create Shipping Label".
7. In the `Ship from` section, tap the "Continue" button.
8. Confirm that the Edit Address screen opens (with the "Name missing" validation error prompting you to enter a name).

## Submitter Checklist

Update release notes:
